TESTS/mbed_drivers/rtc/main.cpp

```diff
@@ -16,41 +16,423 @@
  */
 #include "mbed.h"
 #include "greentea-client/test_env.h"
-#include "unity/unity.h"
-#include "utest/utest.h"
+#include "unity.h"
+#include "utest.h"
+#include "rtos.h"
+#include "rtc_api.h"
 
 using namespace utest::v1;
 
-#define CUSTOM_TIME  1256729737
+/* On some boards RTC counter can not be
+ * Initialised with 0 value in such case
+ * drivers sets RTC counter to 1. */
+#define CUSTOM_TIME_0  1
+#define CUSTOM_TIME_1  1256729737
+#define CUSTOM_TIME_2  2147483645
 
-void test_case_rtc_strftime() {
-    greentea_send_kv("timestamp", CUSTOM_TIME);
+static int rtc_enabled_ret;
+static time_t rtc_time_val;
+static bool rtc_read_called;
+static bool rtc_write_called;
+static bool rtc_init_called;
+static bool rtc_isenabled_called;
 
-    char buffer[32] = {0};
-    char kv_buff[64] = {0};
-    set_time(CUSTOM_TIME);  // Set RTC time to Wed, 28 Oct 2009 11:35:37
+/* Stub of RTC read function. */
+static time_t read_rtc_stub(void)
+{
+    rtc_read_called = true;
 
-    for (int i=0; i<10; ++i) {
-        time_t seconds = time(NULL);
-        sprintf(kv_buff, "[%ld] ", seconds);
-        strftime(buffer, sizeof(buffer), "%Y-%m-%d %H:%M:%S %p", localtime(&seconds));
-        strcat(kv_buff, buffer);
-        greentea_send_kv("rtc", kv_buff);
-        wait(1);
-    }
+    return rtc_time_val;
 }
 
-Case cases[] = {
-    Case("RTC strftime", test_case_rtc_strftime),
-};
+/* Stub of RTC write function. */
+static void write_rtc_stub(time_t t)
+{
+   rtc_write_called = true;
+
+   rtc_time_val = t;
+}
+
+/* Stub of RTC init function. */
+static void init_rtc_stub(void)
+{
+    rtc_init_called = true;
+}
+
+/* Stub of RTC isenabled function. */
+static int isenabled_rtc_stub(void)
+{
+    rtc_isenabled_called = true;
+
+    return rtc_enabled_ret;
+}
+
+/* This test verifies if attach_rtc provides availability to
+ * connect specific RTC driver functions.
+ *
+ * This is unit test to verify if correct functions are used
+ * to support RTC.
+ *
+ * Given specific RTC driver functions have been attached (stubs).
+ * When set_time/time functions are called.
+ * Then set_time/time functions use attached RTC functions.
+ */
+void test_attach_RTC_funtions()
```

Comment:
That's too long. We should try checking one thing in a test (within reason).

Reply:
Spited into two cases. First case verify attaching of stub driver functions and second verify attaching of original RTC driver functions.